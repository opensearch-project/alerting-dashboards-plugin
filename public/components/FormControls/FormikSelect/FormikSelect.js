/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

/*
 *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *   Licensed under the Apache License, Version 2.0 (the "License").
 *   You may not use this file except in compliance with the License.
 *   A copy of the License is located at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   or in the "license" file accompanying this file. This file is distributed
 *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 *   express or implied. See the License for the specific language governing
 *   permissions and limitations under the License.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { EuiSelect } from '@elastic/eui';

import FormikInputWrapper from '../FormikInputWrapper';
import FormikFormRow from '../FormikFormRow';

const FormikSelect = ({
  name,
  formRow = false,
  fieldProps = {},
  rowProps = {},
  inputProps = {},
  selectedOption = undefined,
}) => (
  <FormikInputWrapper
    name={name}
    fieldProps={fieldProps}
    render={({ field, form }) =>
      formRow ? (
        <FormikFormRow name={name} form={form} rowProps={rowProps}>
          <FieldSelect name={name} form={form} field={field} inputProps={inputProps} />
        </FormikFormRow>
      ) : (
        <FieldSelect
          name={name}
          form={form}
          field={field}
          inputProps={inputProps}
          selectedOption={selectedOption}
        />
      )
    }
  />
);

const FieldSelect = ({
  name,
  field,
  form,
  inputProps: { onChange, isInvalid, ...rest },
  selectedOption,
}) => (
  <EuiSelect
    name={name}
    id={name}
    {...rest}
    value={selectedOption ? selectedOption : field.value}
    onChange={(e) =>
      typeof onChange === 'function' ? onChange(e, field, form) : field.onChange(e)
    }
    isInvalid={typeof isInvalid === 'function' ? isInvalid(name, form) : isInvalid}
  />
);

FormikSelect.propTypes = {
  name: PropTypes.string.isRequired,
  formRow: PropTypes.bool,
  fieldProps: PropTypes.object,
  rowProps: PropTypes.object,
  inputProps: PropTypes.object,
};

FieldSelect.propTypes = {
  name: PropTypes.string.isRequired,
  form: PropTypes.object.isRequired,
  inputProps: PropTypes.object.isRequired,
  field: PropTypes.object.isRequired,
};

export default FormikSelect;
