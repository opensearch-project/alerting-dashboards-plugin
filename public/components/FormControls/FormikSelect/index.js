/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import FormikSelect from './FormikSelect';

export default FormikSelect;
