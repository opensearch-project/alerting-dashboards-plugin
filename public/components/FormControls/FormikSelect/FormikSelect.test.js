/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { render } from 'enzyme';
import { Formik } from 'formik';

import FormikSelect from './FormikSelect';

describe('FormikSelect', () => {
  test('renders', () => {
    const component = (
      <Formik>
        <FormikSelect name="testing" inputProps={{ options: [{ value: 'test', text: 'test' }] }} />
      </Formik>
    );

    expect(render(component)).toMatchSnapshot();
  });
});
