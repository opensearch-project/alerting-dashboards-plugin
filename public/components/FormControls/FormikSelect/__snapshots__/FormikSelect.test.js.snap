// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormikSelect renders 1`] = `
<div
  class="euiFormControlLayout euiFormControlLayout--compressed"
>
  <div
    class="euiFormControlLayout__childrenWrapper"
  >
    <select
      class="euiSelect euiSelect--compressed"
      id="testing"
      name="testing"
    >
      <option
        value="test"
      >
        test
      </option>
    </select>
    <div
      class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
    >
      <span
        class="euiFormControlLayoutCustomIcon"
      >
        <div>
          EuiIconMock
        </div>
      </span>
    </div>
  </div>
</div>
`;
